### ISSUE TITLE
Map-box-tile API endpoint throws FileExistError

### ISSUE THREAD

**Issue by htmlboss** [DFO-Ocean-Navigator/Ocean-Data-Map-Project#458]
![image](https://user-images.githubusercontent.com/5572045/60593469-660b4c00-9d7d-11e9-8d63-e18b67177890.png)

**Comment by jamesprayner** [comment]
Weird. The directory creation is wrapped in an if:
```    # Write tile to cache and send file
    if not os.path.isdir(basedir):
      os.makedirs(basedir)
```
The directory that is created is used for multiple tiles (only zoom and x, the filename contains the y coordinate), so probably this is happening when multiple tiles from the same z and x are being requested at the same time (race condition)? Not sure how to approach this.

**Comment by htmlboss** [comment]
Yeah race condition is possible. The exception itself is vague because it may be thrown when a file *or* directory already exists.
